pkg/certificate/conditions.go

```diff
@@ -0,0 +1,111 @@
+	validatingRollback v1alpha2.MeshRootCertificateConditionType = "ValidatingRollback"
+)
+
+const (
+	trueStatus    v1alpha2.MeshRootCertificateConditionStatus = "True"
+	falseStatus   v1alpha2.MeshRootCertificateConditionStatus = "False"
+	unknownStatus v1alpha2.MeshRootCertificateConditionStatus = "Unknown"
+)
+
+const (
+	pendingReason = "Pending"
+
+	// Accepted type
+	certificateAcceptedReason = "CertificateAccepted"
+	errorRetrievingCAReason   = "ErrorRetrievingCA"
+
+	// ValidatingRollout type
+	passiveStateValidatingReason      = "PassiveState"
+	passivelyInUseForValidatingReason = "CertificatePassivelyInUseForValidation"
+	notAcceptedValidatingReason       = "NotAccepted"
+
+	// IssuingRollout type
+	passiveStateIssuingReason      = "PassiveState"
+	passivelyInUseForIssuingReason = "CertificatePassivelyInUseForIssuing"
+	notAcceptedIssuingReason       = "NotAccepted"
+
+	// IssuingRollback type
+	isNotReadyIssuingReason      = "IsNotReady"
+	passiveStateCertIssuedReason = "PassiveStateAndCertIsBeingIssued"
+	noLongerIssuingReason        = "NoLongerIssuing"
+
+	// ValidatingRollback type
+	isNotReadyValidatingReason    = "IsNotReady"
+	passiveStateCertTrustedReason = "PassiveStateAndCertIsTrusted"
+	noLongerValidatingReason      = "NoLongerValidating"
+
+	// Ready type
+	rotationCompleteReason = "RotationComplete"
+)
+
+func setMRCCondition(mrc *v1alpha2.MeshRootCertificate, conditionType v1alpha2.MeshRootCertificateConditionType, conditionStatus v1alpha2.MeshRootCertificateConditionStatus, conditionReason string, message string) {
+	newCondition := v1alpha2.MeshRootCertificateCondition{
+		Type:    conditionType,
+		Status:  conditionStatus,
+		Reason:  string(conditionReason),
+		Message: message,
+	}
+
+	now := metav1.NewTime(time.Now())
+	newCondition.LastTransitionTime = &now
+
+	for i, condition := range mrc.Status.Conditions {
+		if condition.Type != newCondition.Type {
+			continue
+		}
+
+		// only update transition time on status change
+		if condition.Status == newCondition.Status {
+			newCondition.LastTransitionTime = condition.LastTransitionTime
+		} else {
+			log.Debug().Str("mrc", namespacedMRCName(mrc)).Msgf("updating last transition time for condition type %s", condition.Type)
+		}
+
+		// update condition
+		mrc.Status.Conditions[i] = newCondition
+		log.Debug().Str("mrc", namespacedMRCName(mrc)).Msgf("updated existing condition of type %s", condition.Type)
+		return
+	}
+
+	// condition type not found, append it to condition list
+	mrc.Status.Conditions = append(mrc.Status.Conditions, newCondition)
+	log.Debug().Str("mrc", namespacedMRCName(mrc)).Msgf("added new condition of type %s", conditionType)
+}
+
+func getMRCCondition(mrc *v1alpha2.MeshRootCertificate, conditionType v1alpha2.MeshRootCertificateConditionType) *v1alpha2.MeshRootCertificateCondition {
+	for _, condition := range mrc.Status.Conditions {
+		if condition.Type == conditionType {
+			return &condition
```

Comment:
I feel like you probably don't want a pointer here since you care about the underlying value that the condition is set to at the moment, but I could be wrong /cc @steeling for his perspective

Reply:
Is this function only used in the tests?